### ISSUE TITLE
Feature: Brew to PDF

### ISSUE THREAD

**Issue by 5e-Cleric** [naturalcrit/homebrewery#3190]
### Feature description
Because browser print to PDF has a number of problems and incompatibilities with our documents, there has been talk about making our own system to export brews as PDFs

Open Issues this feature could potentially resolve:

#2716
#2312
#2278
#1311
#751
#407

### Minimum Viable Product
As a minimum, this feature should include the following list of capabilities:

- Features included in the system dialog print:
    - Choose which pages to print
    - Layout(portrait/landscape)
    - Paper size
    - Pages per sheet
    - Margins
    - Scale
- Possible new features:
    - Create/keep bookmarks
    - Embed HTML hyperlinks
    - Optimize images

If you have any idea related to this, or that we should add to the MVP, comment in this issue.

**Comment by lucastucious** [comment]
I like to choose color/black and white when I print to pdf !

**Comment by lucastucious** [comment]
Another thing is struggle to have is the dpi resolution. I don't know why, but some pdf printer downscale everything to 76dpi. I use homebrewery to print things for print, so a nice 300dpi is mandatory.

**Comment by 5e-Cleric** [comment]
AFAIK dpi will depend on the printer, but we will try to achieve a dpi setting if we can.

**Comment by dbolack-ab** [comment]
I think we need to add the following to the feature list.

 - [ ] Print on Demand ready - Typically this means PDF output in  PDF/X-1a:2001 or PDF/X-3:2002.

**Comment by calculuschild** [comment]
Note that "Embed HTML hyperlinks" and "Color/B&W option" are already built-in to the basic Chrome print.


https://github.com/naturalcrit/homebrewery/assets/5878534/49f83200-46e2-473b-8678-919d6ddc13c6
![image](https://github.com/naturalcrit/homebrewery/assets/5878534/3e25d637-b3d3-4049-b7b6-84a5076f02bf)

**Comment by Gazook89** [comment]
> I think we need to add the following to the feature list.
>
> Print on Demand ready - Typically this means PDF output in PDF/X-1a:2001 or PDF/X-3:2002.
> 
> This is the requirement for DriveThruRPG and is essentially a passthrough for Lightning Source specs.

[Here is the list of DTRPG requirements for Print on Demand.](https://help.drivethrupartners.com/hc/en-us/articles/12780800178583-Quick-Specifications-for-Print-Books#h_01HFVSQQNE7CNT9CBGDAMQC835).

**Comment by 5e-Cleric** [comment]
Related, this are the normal options to convert a word to pdf, using adobe acrobat:
<img width="286" alt="image" src="https://github.com/naturalcrit/homebrewery/assets/91662974/0b322b53-fc91-4b39-af7b-2bc106e0fb68">
We could consider saving code comments as pdf comments?

**Comment by ericscheid** [comment]
I use code comments `<!-- comment -→` for temporarily (?) disabling content. (and more sensible reasons).

Having those comments appear in the PDF would be confusing. If done, this needs to be optional.

**Comment by 5e-Cleric** [comment]
Perhaps this snippet could help?

```
{{PDFComment
Lorem ipsum dolor sit amet.
}}
```

**Comment by lucastucious** [comment]
One thing that should be great : 
An option to have a "printer friendly" checkbox, and maybe a dedicated css class/file for enabling/desabling graphics.
Quite the same as the "Background Graphics" checkbox, but extendable for .page img, maybe with a whitelist of css classes

**Comment by calculuschild** [comment]
@lucastucious 

There is already a printer friendly button in the **Style Tab** that injects the appropriate CSS into your brew.

**Comment by lucastucious** [comment]
Yeah but this edit the document. I was speaking more of a choice in the print dialog : full pdf OR ink friendly

**Comment by 5e-Cleric** [comment]
We might want to have those as separate features, imagine these possibilities:

- User A has a brew, without print friendly, User B wants to print it, but needs it print friendly, User B needs to clone the brew and add the print friendly option.

- User A has a brew with print friendly option enabled, User B prefers without that, once again, User B needs to clone the brew and remove that.

**Comment by lucastucious** [comment]
That's why I imagine the print friendly as a checkbox

**Comment by dbolack-ab** [comment]
> That's why I imagine the print friendly as a checkbox

Without being able to generate the PDF via our code, rather than a third party service or package, there is no practical way to do this and to date that has proven to be nearly impossible if you want a useful PDF.

**Comment by lucastucious** [comment]
Isn't what we're talking here, making our own way to generate pdf ?

**Comment by dbolack-ab** [comment]
At present, that isn't possible

**Comment by calculuschild** [comment]
Yes, "Print-friendly" could easily be a checkbox that injects the appropriate CSS behind the scenes without actually editing the document contents. The checkbox wouldn't go in the browser print dialog, but it could be a dropdown option under the "Get PDF" button or something. If that option is checked, we apply "Print Friendly" CSS before triggering the print dialog, then remove it when the dialog closes.

**Comment by lucastucious** [comment]
Yes. That's exactly what I was thinking. I know that I could just duplicate my brew and add the print css myself but since we discuss about a printing system, it's worth to talk about a automatic way
